magma/syntax/combinational2.py

```diff
@@ -1,14 +1,45 @@
 import functools
-from typing import Optional
+import libcst as cst
+import libcst.matchers as match
+from typing import Optional, MutableMapping
 
-from ast_tools.passes import apply_passes, ssa, if_to_phi, bool_to_bit
+from ast_tools.common import gen_free_name
+from ast_tools.passes import (apply_passes, ssa, if_to_phi, bool_to_bit, Pass,
+                              PASS_ARGS_T)
 from ast_tools.stack import SymbolTable
 
 from ..circuit import Circuit, IO
 from ..protocol_type import MagmaProtocol
 from ..t import Type
 
 from .util import build_io_args, build_call_args, wire_call_result
+from magma.set_name import set_name
+
+
+class NameSetter(cst.CSTTransformer):
+    def __init__(self, set_name_var):
+        self.set_name_var = set_name_var
+
+    def leave_Assign(self, original_node, updated_node):
```

Comment:
You might be interested in the match decorators.  I have no problem with this though. 

https://github.com/leonardt/ast_tools/blob/master/ast_tools/transformers/normalizers.py#L9
https://libcst.readthedocs.io/en/latest/matchers.html

Reply:
Hmm, nice seems convenient